### PR TITLE
Multiple updates; test passer

### PR DESCRIPTION
**Pull request by ak-2485**
Multiple updates to the traversal phase, the azimuthal/angular hits, and the test cases.
Traversal phase: remove costly array updates in traversal main.
Azimuthal/angular hit: include collinear cases in conditionals, update time comparisons to use function strictly less than.
Test cases: Update test cases based on (a) traversal choices for ambiguities such as ray collinear with boundaries and (b) corrections made by checking 3D vis. in GeoGebra.

### CODE REVIEW

File: matlab/spherical-traversal/angular_hit.m [hunk 1]
```diff
@@ -76,71 +97,95 @@
...
+% Compute inverses of slopes
 if ~approximatelyEqual(ray_direction(1),0.0,1e-12,1e-8)
     inv_dir = 1/ray_direction(1);
 else
     inv_dir = 1/ray_direction(2);
 end
-
-if intersect_max == 1
+% Compute intersection times
+if intersect_max == 1 || collinear_max == 1
```
Comment (cgyurgyik) [inline matlab/spherical-traversal/angular_hit.m:107]: so ```t_max``` is initialized if either of these are 1. What is its value if neither of these are 1?

Comment (ak-2485) [reply matlab/spherical-traversal/angular_hit.m:107]: They should probably be initialized to t_end.

File: matlab/spherical-traversal/azimuthal_hit.m [hunk 2]
```diff
@@ -76,71 +97,95 @@
...
+            strictlyLess(t_min,t_end,1e-12,1e-8)
         tMaxPhi = t_max;
          if verbose
             fprintf("hit origin t\n")
+         end
+         pert_t = 0.1;
+         pert_x = sphere_center(1) + ray_direction(1) * pert_t;
+         pert_z = sphere_center(3) + ray_direction(3) * pert_t;
+         a = sphere_center(1) - pert_x;
+         b = sphere_center(2) - pert_z;
```
Comment (cgyurgyik) [inline matlab/spherical-traversal/azimuthal_hit.m:146]: Should this be: ``` b = sphere_center(3) - pert_z;```

File: matlab/spherical-traversal/angular_hit.m (hunk 1 above)
Comment (cgyurgyik) [reply matlab/spherical-traversal/angular_hit.m:107]: Ok I leave mine defaulted to 0.0, and only change in this case. Either 0.0 or t_end should work though.

**Comment by cgyurgyik** [comment]
IMO, the use of ```some_value == 1``` decreases readability. Instead, use booleans.

```
if intersect_min == 1
     ...
end
```
versus
```
if is_intersect_min
    ...
end
```

File: matlab/spherical-traversal/azimuthal_hit.m (hunk 2 above)
Comment (ak-2485) [reply matlab/spherical-traversal/azimuthal_hit.m:146]: Yes.

**Pull request merged**